src/main/resources/com/axis/system/jenkins/plugins/downstream/yabv/BuildFlowAction/buildFlow.groovy

```diff
@@ -14,74 +14,118 @@ import static com.axis.system.jenkins.plugins.downstream.tree.Matrix.Arrow
+      return
+    }
+
+    Set<Job> jobs = matrix.cellDataAsSet.collect { data ->
+      getJob(data)
+    }.toSet()
+
+    NameNormalizer nameNormalizer = getNameNormalizer(jobs)
+
+    matrix.get().each { row ->
+      row.each { cell ->
+        drawArrow(cell?.arrow)
+        drawCellData(cell?.data, nameNormalizer, options)
       }
-      gridCoords.col++
     }
   }
 }
 
-private void drawCellData(CssGridCoordinates gridCoords, Object data, NameNormalizer
-    nameNormalizer, BuildFlowOptions options) {
+private static Job getJob(Object data) {
   if (data instanceof Run) {
-    drawBuildInfo(gridCoords, data, nameNormalizer, options)
+    return data.parent
+  } else if (data instanceof Queue.Item && data.task instanceof Job) {
+    return (Job) data.task
+  }
+  return null
+}
+
+private static NameNormalizer getNameNormalizer(Set<Job> jobs) {
+  return new NameNormalizer(jobs, {
+    it.displayName
+  }, {
+    it instanceof Item ? it.parent : null
+  })
+}
+
+private void drawCellData(Object data, NameNormalizer nameNormalizer, BuildFlowOptions options) {
+  if (data instanceof Run) {
+    drawBuildInfo(data, nameNormalizer, options)
   } else if (data instanceof Queue.Item) {
-    drawQueueItemInfo(gridCoords, data, nameNormalizer)
+    drawQueueItemInfo(data, nameNormalizer, options)
+  } else {
+    div { }
   }
 }
 
-private void drawBuildInfo(CssGridCoordinates gridCoords, Run build, NameNormalizer
-    nameNormalizer, BuildFlowOptions options) {
+private void drawBuildInfo(Run build, NameNormalizer nameNormalizer, BuildFlowOptions options) {
   def color = build.iconColor
   def colorClasses = color.name().replace('_', ' ') + ' ' + (build == my.target ? 'SELECTED' : '')
-  div(class: "build-info ${colorClasses}",
-      style: gridCoords.cssStyleString) {
+  div(class: "build-info ${colorClasses} ${options.flattenView ? 'FLAT' : ''}") {
     a(class: 'model-link inside', href: "${rootURL}/${build.url}") {
-      span("${nameNormalizer.getNormalizedName(build.parent)} ${build.displayName}")
+      if (options.flattenView) {
+        span("${build.displayName}")
+      } else {
+        span("${nameNormalizer.getNormalizedName(build.parent)} ${build.displayName}")
+      }
     }
     if (options.showDurationInfo) {
       span(class: 'duration-info', build.durationString)
     }
-    if (options.showBuildHistory) {
-      div(class: "build-flow-build-history") {
+    if (options.showBuildHistory && !my.buildFlowOptions.flattenView) {
+      div(class: 'build-flow-build-history') {
         currentBuild = build.previousBuild
         for (int i = 0; i < 5 && currentBuild != null; i++) {
           a(href: "${rootURL}/${currentBuild.url}") {
             def currentColor = currentBuild.iconColor
-            div(class: "build-flow-build-history-dot build-info ${currentColor.name().replace('_', ' ')}",
+            div(class: 'build-flow-build-history-dot build-info ' +
+                currentColor.name().replace('_', ' '),
```

Comment:
Maybe time to break out color.name().replace('_', ' ') to a helper method, duplicated in a few places.